### PR TITLE
doc(zjow): polish the notation of classes and functions in torch_utils and utils

### PR DESCRIPTION
**Pull request by zjowowen**

### CODE REVIEW

File: ding/torch_utils/dataparallel.py [hunk 1]
```diff
@@ -3,10 +3,31 @@
...
+        """
         super().__init__(module, device_ids=None, output_device=None, dim=0)
         self.module = module
 
     def parameters(self, recurse: bool = True):
+        """
+        Overview:
+            Return the parameters of the module.
+        Arguments:
+            - recurse (:obj:`bool`): Whether to return the parameters of the submodules.
```
Comment (PaParaZz1) [inline ding/torch_utils/dataparallel.py:31]: add Returns and typing lint in this class

File: ding/torch_utils/network/diffusion.py [hunk 2]
```diff
@@ -234,7 +360,7 @@ def __init__(
     ) -> None:
         """
         Overview:
-            temporal net
+            initialization of DiffusionUNet1d class
```
Comment (PaParaZz1) [inline ding/torch_utils/network/diffusion.py:363]: Initialization

File: ding/torch_utils/network/merge.py [hunk 3]
```diff
@@ -94,13 +114,20 @@ class TorchBilinearCustomized(nn.Module):
...
         Customized Torch Bilinear implementation.
-    Arguments:
-        - in1_features (:obj:`int`): The size of each first input sample.
-        - in2_features (:obj:`int`): The size of each second input sample.
-        - out_features (:obj:`int`): The size of each output sample.
+    Interfaces:
+        ``__init__``, ``forward``
     """
 
     def __init__(self, in1_features, in2_features, out_features):
```
Comment (PaParaZz1) [inline ding/torch_utils/network/merge.py:121]: add typing lint for this file

File: ding/utils/normalizer_helper.py [hunk 4]
```diff
@@ -190,11 +233,23 @@ class CDFNormalizer(Normalizer):
...
+            Initialize the CDFNormalizer object.
+        Arguments:
+            - X (:obj:`np.ndarray`): The data to be normalized.
+        """
+
         super().__init__(atleast_2d(X))
         self.dim = self.X.shape[1]
         self.cdfs = [CDFNormalizer1d(self.X[:, i]) for i in range(self.dim)]
 
     def __repr__(self):
```
Comment (PaParaZz1) [inline ding/utils/normalizer_helper.py:247]: add Returns

File: ding/utils/data/dataset.py [hunk 5]
```diff
@@ -100,31 +171,76 @@ def _cal_statistics(self, dataset, env, eps=1e-3, add_action_buffer=True):
...
+        """
+
         return DatasetStatistics(mean=self.mean, std=self.std, action_bounds=self.action_bounds)
 
 
 @DATASET_REGISTRY.register('hdf5')
 class HDF5Dataset(Dataset):
+    """
+    Overview:
+        HDF5 dataset, which is used for offline RL algorithms.
```
Comment (PaParaZz1) [inline ding/utils/data/dataset.py:226]: add some explanation about hdf5 format

File: ding/utils/data/dataset.py [hunk 6]
```diff
@@ -166,12 +294,26 @@ def __getitem__(self, idx: int) -> Dict[str, torch.Tensor]:
...
+        Arguments:
+            - dataset (:obj:`Dict[str, np.ndarray]`): The dataset.
+        """
+
         self._data = {}
         for k in dataset.keys():
             logging.info(f'Load {k} data.')
             self._data[k] = dataset[k][:]
 
     def _cal_statistics(self, eps=1e-3):
```
Comment (PaParaZz1) [inline ding/utils/data/dataset.py:309]: typing lint

File: ding/utils/data/dataset.py [hunk 7]
```diff
@@ -726,26 +970,72 @@ def _load_buffer(self, suffix):
...
+        """
+
         buffer_index = np.random.randint(self._num_replay_buffers)
         return self._replay_buffers[buffer_index].sample_transition_batch(batch_size=batch_size, indices=indices)
 
 
 class PCDataset(Dataset):
+    """
+    Overview:
+        Dataset for pre-computed BFS.
```
Comment (PaParaZz1) [inline ding/utils/data/dataset.py:996]: Procedure Cloning

File: ding/utils/data/dataset.py [hunk 8]
```diff
@@ -807,32 +1097,83 @@ def load_env(seed):
 
 @DATASET_REGISTRY.register('bco')
 class BCODataset(Dataset):
+    """
+    Overview:
+        Dataset for BCO.
```
Comment (PaParaZz1) [inline ding/utils/data/dataset.py:1102]: add the whole name of BCO

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: ding/torch_utils/loss/multi_logits_loss.py [hunk 9]
```diff
@@ -88,7 +116,7 @@ def has_augmented_path(t, binary_distance_matrix):
             while True:
                 visx.fill(False)
                 visy.fill(False)
-                distance_matrix = get_distance_matrix(lx, ly, mat, M)
+                distance_matrix = self._get_distance_matrix(lx, ly, mat, M)
```
Comment (zjowowen) [inline ding/torch_utils/loss/multi_logits_loss.py:119]: @static_method

File: ding/torch_utils/dataparallel.py (hunk 1 above)
Comment (zjowowen) [reply ding/torch_utils/dataparallel.py:31]: fixed.

File: ding/torch_utils/network/diffusion.py (hunk 2 above)
Comment (zjowowen) [reply ding/torch_utils/network/diffusion.py]: fixed.

File: ding/torch_utils/network/merge.py (hunk 3 above)
Comment (zjowowen) [reply ding/torch_utils/network/merge.py]: fixed.

File: ding/utils/normalizer_helper.py (hunk 4 above)
Comment (zjowowen) [reply ding/utils/normalizer_helper.py]: fixed.

File: ding/utils/data/dataset.py (hunk 6 above)
Comment (zjowowen) [reply ding/utils/data/dataset.py]: fixed.

File: ding/utils/data/dataset.py (hunk 7 above)
Comment (zjowowen) [reply ding/utils/data/dataset.py]: fixed.

File: ding/utils/data/dataset.py (hunk 8 above)
Comment (zjowowen) [reply ding/utils/data/dataset.py]: fixed.

File: ding/torch_utils/loss/contrastive_loss.py [hunk 10]
```diff
@@ -45,26 +45,34 @@ def __init__(
...
+        self._y_encoder = self._create_encoder(y_size, heads[1])
         self._temperature = temperature
 
-    def _get_encoder(self, obs: Union[int, SequenceType], heads: int) -> nn.Module:
+    def _create_encoder(self, obs_size: Union[int, SequenceType], heads: int) -> nn.Module:
+        """
+        Overview:
+            Create the encoder for the input obs.
+        Arguments:
+            - obs_size (:obj:`Union[int, SequenceType]`): input shape for x, both the obs shape and the encoding shape \
```
[the remark this answers is not in the document]
Comment (zjowowen) [reply ding/torch_utils/loss/contrastive_loss.py:57]: fixed.

File: ding/torch_utils/loss/multi_logits_loss.py (hunk 9 above)
Comment (zjowowen) [reply ding/torch_utils/loss/multi_logits_loss.py:119]: fixed.

**Review by PaParaZz1 — APPROVED** [review]
